base/compiler/typeutils.jl

```diff
@@ -231,8 +231,7 @@ function unionsplitcost(argtypes::Union{SimpleVector,Vector{Any}})
     nu = 1
     max = 2
     for ti in argtypes
-        # TODO remove this to implement callsite refinement of MustAlias
-        if isa(ti, MustAlias) && isa(widenconst(ti), Union)
+        if !isvarargtype(ti)
             ti = widenconst(ti)
```

Comment:
This `widenconst` would be wasteful in the native compilation pipeline since we don't enable `MustAlias` by default. Maybe we want to add a hook `has_extended_unionsplit(𝕃::AbstractLattice)` to configure this widening?

Reply:
sure, seems reasonable